src/UI/IdentityBuilderUIExtensions.cs

```diff
@@ -36,6 +36,11 @@ public static IdentityBuilder AddDefaultUI(this IdentityBuilder builder)
                     .MakeGenericType(builder.UserType));
             builder.Services.TryAddTransient<IEmailSender, EmailSender>();
 
+            if (builder.UserType == typeof(IdentityUser))
+            {
+                builder.Services.AddSingleton<IUIUserInstanceFactory<IdentityUser>, DefaultUIUserInstanceFactory>();
```

Comment:
We can be a bit smarter than this too I think, with `IdentityUserFactory<TUser, TKey> : IUserFactory<TUser> where TUser : IdentityUser<TKey>` and use reflection on builder.UserType to see if its a derived IdentityUser and then register the appropriate closed IdentityUserFactory.